src/components/Feed/CreatePost.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from "react";
+import { api } from "@/utils/api";
+import { useSession } from "next-auth/react";
+import Image from "next/image";
+
+export default function CreatePost() {
+  const { data } = useSession() || "";
+  const [post, setPost] = useState({
+    content: "",
+    groupId: undefined,
+  });
+  const mutation = api.posts.createPost.useMutation({});
```

Comment:
in the router root the route is named "post" but here it is being called with "posts". One or the either needs to change.